sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -188,23 +199,39 @@ def tune(
         pip_index_url: str = "https://pypi.org/simple",
         metrics_collector_config: Dict[str, Any] = {"kind": "StdOut"},
     ):
-        """Create HyperParameter Tuning Katib Experiment from the objective function.
+        """Create HyperParameter Tuning Katib Experiment using one of the following options:
+        - External models and datasets: Specify both `model_provider_parameters` and `dataset_provider_parameters` to download models and datasets from external platforms (currently supports HuggingFace and Amazon S3) using the Storage Initializer. The `trainer_parameters` should be of type `HuggingFaceTrainerParams` to set the hyperparameters search space. This API will automatically define the "Trainer" class in HuggingFace with the provided parameters and utilize `Trainer.train()` from HuggingFace to obtain the metrics for optimizing hyperparameters. 
+        - Custom objective function: Specify the `objective` parameter to define your own objective function. The `base_image` parameter will be used to execute the objective function. `trainer_parameters` should be a dictionary to define the search space for these parameters.
 
         Args:
             name: Name for the Experiment.
+            model_provider_parameters: Parameters for the model provider in the Storage Initializer.
+                For example, HuggingFace model name and Transformer type for that model, like: AutoModelForSequenceClassification. This argument must be the type of `kubeflow.storage_initializer.hugging_face.HuggingFaceModelParams`.
```

Comment:
Can we add the example for S3 as well here?

Reply:
Thank you for your suggestions! Do you mean add a use example of S3?

Reply:
We don't support S3 as model provider today.

Reply:
I'm not sure about this. @andreyvelich Do you have any idea about that?